### PR TITLE
Upgrade Bevy to 0.15

### PR DESCRIPTION
**Pull request by snendev**
# Issue

This upgrades Bevy to 0.15.

# Changes

- Included the required `bevy_window` feature in dependencies
- Renamed `App::observe` to `App::add_observer`
- Renamed `Condition(?)::and_then` to `::and`
- Added `repeat` field to keyboard inputs
- `World::flush_commnads` is now private, replaced with `World::flush`
- Updated `gamepad` example to accommodate refactors around (1) gamepads (now entities instead of tracked by resources) and (2) Text. This comprises the bulk of the meaningful changes.

Hope this helps!

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
Fantastic. Thanks for all your help here; I've sent you an invite for maintainer privileges on Github for this crate. I'm not actively using it, so it's nice to share the load and get real user feedback.

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
Changes look good, but this needs release notes. Can you bump the crate version number in this PR too so we can release right after this is merged?

**Comment by snendev** [comment]
🤦🏻 my bad, I've forgotten release notes every time. can do!

**Review by alice-i-cecile — APPROVED** [review]

**Pull request merged**
